bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQBridgeHandler.java

```diff
@@ -0,0 +1,375 @@
+        if (devicePollingJob != null) {
+            devicePollingJob.cancel(true);
+        }
+        tokenManager.cleanupTokenRegistry(
+                Objects.requireNonNull(getBridge(), "Not expected bridge null here").getUID().getId());
+        super.handleRemoval();
+    }
+
+    @Override
+
+    public void dispose() {
+        ScheduledFuture<?> devicePollingJob = this.devicePollingJob;
+        if (devicePollingJob != null) {
+            devicePollingJob.cancel(true);
+            this.devicePollingJob = null;
+        }
+    }
+
+    @Override
+    public <T> T getConfigAs(Class<T> configurationClass) {
+        return super.getConfigAs(configurationClass);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing LGThinq bridge handler.");
+        lgthinqConfig = getConfigAs(LGThinQBridgeConfiguration.class);
+        lgDevicePollingRunnable.lgthinqConfig = lgthinqConfig;
+        if (lgthinqConfig.username.isEmpty() || lgthinqConfig.password.isEmpty() || lgthinqConfig.language.isEmpty()
+                || lgthinqConfig.country.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/error.mandotory-fields-missing");
+        } else {
+            // updateStatus(ThingStatus.UNKNOWN);
+            startLGThinqDevicePolling();
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("Bridge Configuration was updated. Cleaning the token registry file");
+        File f = new File(String.format(THINQ_CONNECTION_DATA_FILE, getThing().getUID().getId()));
+        if (f.isFile()) {
+            // file exists. Delete it
+            if (!f.delete()) {
+                logger.error("Unexpected error deleting file:{}", f.getAbsolutePath());
+            }
+        }
+        super.handleConfigurationUpdate(configurationParameters);
+    }
+
+    private void startLGThinqDevicePolling() {
+        // stop current scheduler, if any
+        ScheduledFuture<?> devicePollingJob = this.devicePollingJob;
+        if (devicePollingJob != null && !devicePollingJob.isDone()) {
+            devicePollingJob.cancel(true);
+        }
+        long poolingInterval;
+        int configPollingInterval = lgthinqConfig.getPollingIntervalSec();
+        // It's not recommended to polling for resources in LG API short intervals to do not enter in BlackList
+        if (configPollingInterval < 300 && configPollingInterval != 0) {
+            poolingInterval = TimeUnit.SECONDS.toSeconds(300);
+            logger.info("Wrong configuration value for pooling interval. Using default value: {}s", poolingInterval);
+        } else {
+            if (configPollingInterval == 0) {
+                logger.info("LG's discovery pooling disabled (configured as zero)");
+                return;
+            }
+            poolingInterval = configPollingInterval;
+        }
```

Comment:
This can be simplified, also fixed typo's and log level
```suggestion
        if (configPollingInterval == 0) {
            logger.debug("LG's discovery polling disabled");
            return;
        } else (configPollingInterval < 300){
            configPollingInterval = 300;
            logger.warn("Wrong configuration value for polling interval. Using default value: {}s", configPollingInterval);
        }
        pollingInterval = configPollingInterval;
```